components/director/internal/tenantfetchersvc/subscriber.go

```diff
@@ -141,14 +148,25 @@ func (s *subscriber) createLabel(ctx context.Context, tenant string, runtime *mo
 	})
 }
 
-func (s *subscriber) updateLabel(ctx context.Context, tenant string, runtime *model.Runtime, label *model.Label, labelNewValue []string) error {
-	return s.labelSvc.UpdateLabel(ctx, tenant, label.ID, &model.LabelInput{
-		Key:        s.ConsumerSubaccountIDsLabelKey,
-		Value:      labelNewValue,
-		ObjectType: model.RuntimeLabelableObject,
-		ObjectID:   runtime.ID,
-		Version:    label.Version,
-	})
+func (s *subscriber) updateLabelWithRetry(ctx context.Context, tenant string, runtime *model.Runtime, label *model.Label, labelNewValue []string) error {
+	err := retry.Do(func() error {
```

Comment:
I think you can return directly here:
`return retry.Do(...)`
and delete lines 165-169.